### DISCUSSION TITLE
How to implement deletion of an instance image from a set of series

### DISCUSSION THREAD

**Discussion by ashishsinghintero** [fo-dicom/fo-dicom#1233] [Q&A]
Hi Everyone,
  Thanks in advance, I am trying to implement a scenario, where I have send some dicom images which are part of a series to SCP server from my client. Now I have case where I want to delete a specific dicom image file from SCP server and i want to send this delete request from client.
   If someone is able to share some code sample for this implementation like how to send delete request and receive the response from server, like it's successful or not.

 I have searched lot of places but did not got any proper answer. Please suggest me some solution for it.

**Comment by gofal** [comment]
you cannot delete dicom files from a server via DICOM methods.
But there is IHE, you should look at this. while DICOM and HL7 are standards that define what files and messages have to look like, IHE tries to define workflows which actor has to send which message on which event.
https://www.ihe.net/uploadedFiles/Documents/Radiology/IHE_RAD_TF_Vol1.pdf
In IHE there is a workflow Imaging Object Change Management (IOCM). There it describes, that after a image has been stored, someone can create a small structured report file, referencing the file to be deleted. This structured report is then sent to the archive, which then forwards this to all other viewer/nodes (RAD-66). The archive should then treat those files as deleted (but does not actually really delete it). so in future queries those image should not be included in answers etc.

You see, deleting will not work in every PACS, but only in those, who are supporting IHE IOCM.
There is an annual Connectathon, where vendors meet to validate their implementations. According to that, at least those vendors support IHE IOCM (https://connectathon-results.ihe.net/view_result.php?rows=company&columns=actor&title=integration_profile):

Vendor | Image Manager | Image Display | Change Requester | Imaging Document Source
-- | -- | -- | -- | --
AGFA Healthcare | * |   | * |  
Carestream Health (Kodak) | * |   | * |  
CHILI Gmbh | * | * | * |  
Dell | * |   | * |  
Fujifilm | * |   | * |  
GE Precision Healthcare | * |   | * |  
INFINITT | * |   |   |  
ITH icoserve technology for healthcare Gmbh | * |   | * |  
J4Care Gmbh | * |   | * |  
MedicalCommunications | * |   | * |  
Medweb | * |   |   |  
Paxerahealth | * | * | * |  
Roga-Delft |   |   | * |  
RVC bv | * |   | * |  
Sectra | * |   | * |  
Siemens Healthcare |  |   | * |  
synedra information technologies Gmbh | * |   | * |  
TeraMedica, Inc | * |   |   | *
Tiani "Spirit" Gmbh |  | * |   |  
Visage Imaging | * | * | * |  
Visbion Ltd | * |   | * |  
VISUS Health IT Gmbh | * | * | * |  

But I assume, some vendors require some license fee, to activate their IHE profiles.
